sqlmesh/core/snapshot/definition.py

```diff
@@ -672,6 +685,7 @@ def categorize_as(self, category: SnapshotChangeCategory) -> None:
         )
         if is_forward_only and self.previous_version:
             self.version = self.previous_version.data_version.version
+            self.physical_schema_ = self.previous_version.physical_schema
```

Comment:
@tobymao the fact that we didn't inherit the previous physical schema on forward-only / non-breaking was a bug which is now fixed.